conans/client/build/cmake.py

```diff
@@ -47,18 +48,23 @@ def __init__(self, conanfile, generator=None, cmake_system_name=True,
         self._cmake_system_name = cmake_system_name
         self._make_program = make_program
 
-        self.definitions = self._def_builder.get_definitions()
         self._build_type = self._settings.get_safe("build_type")
         if build_type and build_type != self._build_type:
             # Call the setter to warn and update the definitions if needed
             self.build_type = build_type
+        self._verbose = None
 
     @property
-    def _def_builder(self):
-        return CMakeDefinitionsBuilder(self._conanfile, cmake_system_name=self._cmake_system_name,
-                                       make_program=self._make_program, parallel=self.parallel,
-                                       generator=self.generator,
-                                       set_cmake_flags=self._set_cmake_flags)
+    def definitions(self):
+        builder = CMakeDefinitionsBuilder(self._conanfile, cmake_system_name=self._cmake_system_name,
+                                          make_program=self._make_program, parallel=self.parallel,
+                                          generator=self.generator,
+                                          set_cmake_flags=self._set_cmake_flags)
+        definitions = builder.get_definitions()
+        definitions.update(build_type_definition(self._build_type, self.generator))
+        if self._verbose is not None:
+            definitions.update(verbose_definition(self._verbose))
+        return definitions
```

Comment:
(@lasote) being a `property` it will always have up-to-date values, but on the other hand, you cannot modify the dictionary and get the class members updated (see deleted tests here: https://github.com/conan-io/conan/pull/3450/files#diff-cd58cd0dfc3250b46311d65eb7f413d0L835). But I think this is much better.